### ISSUE TITLE
Rounding error when setting celsius temperature

### ISSUE THREAD

**Issue by ticky** [sampsyo/hass-smartthinq#14]
Hi there,

I just set this up with my LG LP1419IVSM, and I managed to get it working after some tinkering, however, initially it was failing due to a failing temperature conversion - some part of the Home Assistant stack was setting a temperature to 78.80000000000001ºF, and this crashed upon being passed to `_ac.set_fahrenheit`, because the mappings only contain fahrenheit values for each whole degree (see below).

I’ve overcome this temporarily by editing the source to set `fahrenheit` to `False` in `LGDevice`’s `__init__` method, which appears to cause Home Assistant to send the unaltered Celsius value, but this seems like it should be gracefully handled in the default code, given it’s a natural behaviour within Home Assistant.

A partial log from when I experienced this failure is below:

```
2019-05-15 10:21:07 INFO (Thread-20) [custom_components.smartthinq.climate] Setting temperature to 24.5…
  File "/usr/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/homeassistant/.homeassistant/custom_components/smartthinq/climate.py", line 189, in set_temperature
    self._ac.set_fahrenheit(temperature)
  File "/srv/homeassistant/lib/python3.5/site-packages/wideq.py", line 870, in set_fahrenheit
    self.set_celsius(self.f2c[f])
KeyError: 78.80000000000001
```

For further reference, my Air Conditioner’s reported `f2c` mapping is:

```python3
{50: 10, 51: 10.5, 52: 11, 53: 11.5, 54: 12, 55: 12.5, 56: 13, 57: 13.5, 58: 14, 59: 15, 60: 16, 61: 16.5, 62: 17, 63: 17.5, 64: 18, 65: 18.5, 66: 19, 67: 19.5, 68: 20, 69: 20.5, 70: 21, 71: 21.5, 72: 22, 73: 22.5, 74: 23, 75: 23.5, 76: 24, 77: 25, 78: 26, 79: 26.5, 80: 27, 81: 27.5, 82: 28, 83: 28.5, 84: 29, 85: 29.5, 86: 30, 87: 30.5, 88: 31, 89: 31.5, 90: 32, 91: 32.5, 92: 33, 93: 33.5, 94: 34, 95: 35, 96: 36, 97: 36.5, 98: 37, 99: 37.5, 100: 38, 101: 38.5, 102: 39, 103: 39.5, 104: 40}
```

And the `c2f` (noting that this one in fact has every half degree!) is:

```python3
{10: 50, 11: 52, 12: 54, 13: 56, 14: 58, 15: 59, 16: 60, 17: 62, 18: 64, 19: 66, 20: 68, 21: 70, 22: 72, 23: 74, 24: 76, 25: 77, 26: 78, 27: 80, 28: 82, 29: 84, 30: 86, 31: 88, 32: 90, 33: 92, 34: 94, 35: 95, 36: 96, 37: 98, 38: 100, 39: 102, 40: 104, 10.5: 51, 11.5: 53, 12.5: 55, 13.5: 57, 14.5: 58, 15.5: 59, 16.5: 61, 17.5: 63, 18.5: 65, 19.5: 67, 20.5: 69, 21.5: 71, 22.5: 73, 23.5: 75, 24.5: 76, 25.5: 77, 26.5: 79, 27.5: 81, 28.5: 83, 29.5: 85, 30.5: 87, 31.5: 89, 32.5: 91, 33.5: 93, 34.5: 94, 35.5: 95, 36.5: 97, 37.5: 99, 38.5: 101, 39.5: 103}
```

**Comment by sampsyo** [comment]
Makes sense! For what it's worth, you have stumbled across my least favorite part of this whole LG API mess: the f2c and c2f mappings are _not algorithmically calculated_, and you really do have to use those exact lookup tables to translate between the two. Someone at LG decided that they needed their own, in-house, LG-branded versions of these temperature scales. 🙄

I think the thing to do here is to make `LGDevice` here somehow sensitive to the Home Assistant configuration—i.e., it should set `fahrenheit` according to some global HA parameter that decides whether you prefer C or F. I hope this should just require some digging around in the HA source/docs to find the right parameter to use. By adapting our component's configuration to match HA, I assume that HA won't try to automatically convert C to F for you, which is what caused these "misses" in the lookup tables.

**Comment by sampsyo** [comment]
Fixed in #21.
